ioctl/newcmd/node/nodedelegate.go

```diff
@@ -0,0 +1,291 @@
+						message.Determined = false
+						fmt.Println(message.String())
+						return nil
+					} else if ok {
+						return output.NewError(output.APIError, sta.Message(), nil)
+					}
+					return output.NewError(output.NetworkError, "failed to invoke ReadState api", err)
+				}
+				message.Determined = true
+				var ABPs state.CandidateList
+				if err := ABPs.Deserialize(abpResponse.Data); err != nil {
+					return output.NewError(output.SerializationError, "failed to deserialize active BPs", err)
+				}
+
+				bpResponse, err := apiServiceClient.ReadState(
+					ctx,
+					&iotexapi.ReadStateRequest{
+						ProtocolID: []byte("poll"),
+						MethodName: []byte("BlockProducersByEpoch"),
+						Arguments:  [][]byte{byteutil.Uint64ToBytes(epochNum)},
+					},
+				)
+
+				if err != nil {
+					sta, ok := status.FromError(err)
+					if ok {
+						return output.NewError(output.APIError, sta.Message(), nil)
+					}
+					return output.NewError(output.NetworkError, "failed to invoke ReadState api", err)
+				}
+				var BPs state.CandidateList
+				if err := BPs.Deserialize(bpResponse.Data); err != nil {
+					return output.NewError(output.SerializationError, "failed to deserialize BPs", err)
+				}
+				isActive := make(map[string]bool)
+				for _, abp := range ABPs {
+					isActive[abp.Address] = true
+				}
+				aliases := alias.GetAliasMap()
+				for rank, bp := range BPs {
+					votes := big.NewInt(0).SetBytes(bp.Votes.Bytes())
+					message.Delegates = append(message.Delegates, delegate{
+						Address: bp.Address,
+						Rank:    rank + 1,
+						Alias:   aliases[bp.Address],
+						Active:  isActive[bp.Address],
+						Votes:   util.RauToString(votes, util.IotxDecimalNum),
+					})
+				}
+				fmt.Println(message.String())
+			} else {
+				apiServiceClient, err := c.APIServiceClient(ioctl.APIServiceConfig{
+					Endpoint: endpoint,
+					Insecure: insecure,
+				})
+				if err != nil {
+					return err
+				}
+				if epochNum == 0 {
+					chainMetaResponse, err := apiServiceClient.GetChainMeta(context.Background(), &iotexapi.GetChainMetaRequest{})
+					if err != nil {
+						return output.NewError(0, "failed to get chain meta", err)
+					}
+					epochNum = chainMetaResponse.ChainMeta.Epoch.Num
+				}
+
+				response, err := apiServiceClient.GetEpochMeta(context.Background(), &iotexapi.GetEpochMetaRequest{EpochNumber: epochNum})
+
+				if err != nil {
+					return output.NewError(0, "failed to get epoch meta", err)
+				}
+				epochData := response.EpochData
+				aliases := alias.GetAliasMap()
+				message := delegatesMessage{
+					Epoch:       int(epochData.Num),
+					StartBlock:  int(epochData.Height),
+					TotalBlocks: int(response.TotalBlocks),
+				}
+
+				kickoutListRes, err := apiServiceClient.ReadState(context.Background(), &iotexapi.ReadStateRequest{
+					ProtocolID: []byte("poll"),
+					MethodName: []byte("KickoutListByEpoch"),
+					Arguments:  [][]byte{byteutil.Uint64ToBytes(epochNum)},
+				})
+				if err != nil {
```

Comment:
need to handle that in case of error status == codes.NotFound, should not return error but just deserialize empty blacklist because of before easter